ci: remove k8s 1.30 tests check

iPraveenParihar opened the pull request:
<!-- Please take a look at our [Contributing](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#Code-contribution-workflow)
documentation before submitting a Pull Request!
Thank you for contributing to ceph-csi! -->

# Describe what this PR does #

k8s 1.30 tests were excluded for release-v3.11 but
mergify has status check condition for k8s 1.30 tests.
This commit removes these checks from mergify.

## Related issues ##

Closes: #4625 

**Checklist:**

* [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages).
* [x] Reviewed the developer guide on [Submitting a Pull Request](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#development-workflow)
* [x] [Pending release notes](https://github.com/ceph/ceph-csi/blob/devel/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next major release.
* [x] Documentation has been updated, if necessary.
* [x] Unit tests have been added, if necessary.
* [x] Integration tests have been added, if necessary.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

Rakshith-R commented on .mergify.yml:
```diff
@@ -37,15 +37,12 @@ queue_rules:
               - "status-success=ci/centos/k8s-e2e-external-storage/1.27"
               - "status-success=ci/centos/k8s-e2e-external-storage/1.28"
               - "status-success=ci/centos/k8s-e2e-external-storage/1.29"
-              - "status-success=ci/centos/k8s-e2e-external-storage/1.30"
```
There's one more set here https://github.com/ceph/ceph-csi/blob/c96e05a4c257ebafbf587a9f7f0ac0f991585cb4/.mergify.yml#L268-L294

Madhu-1 replied:
instead of this one i would like to run 1.30 test as we might need to do minor releases due to bug fixes.

iPraveenParihar replied:
okay, then I'll include the 1.30 tests for `release-v3.11`

Madhu-1 commented on .github/workflows/pull-request-commentor.yaml:
```diff
@@ -27,10 +27,6 @@ jobs:
           - k8s: "1.26"
             branch: "devel"
 
-          # Ceph-CSI <= 3.11 was released before Kubernetes 1.30
-          - k8s: "1.30"
-            branch: "release-v3.11"
-
```
Sorry for a mistake from my side, now i see that we don't support 1.30 with even patch releases. can you please remove 1.30 check from mergify.yaml where we have `base~=^(release-.+)$` check and nothing else.

iPraveenParihar committed:
ci: remove k8s 1.30 tests from release-v3.11

k8s 1.30 tests were excluded for release-v3.11 but
merigfy had 1.30 tests conditions for release-v3.11.
This commit remove the these conditions for v3.11.

Signed-off-by: Praveen M <m.praveen@ibm.com>

Rakshith-R reviewed (approved):
Thanks

iPraveenParihar commented:
We also need to backport this to v3.11

Madhu-1 commented:
@iPraveenParihar mergify rules are checked on only main/devel branch, if required we can backport it later.

Madhu-1 commented:
@mergifyio queue
